src/UI/Components/ServiceInstanceForm/Components/SuggestionsPopover.tsx

```diff
@@ -0,0 +1,198 @@
+  const handleMenuKeys = (event) => {
+    // If the focus is on the input and the autocomplete menu is open.
+    if (
+      isOpen &&
+      reference.current === event.target &&
+      autocompleteRef &&
+      autocompleteRef.current
+    ) {
+      switch (event.key) {
+        // the up and down arrow keys navigate the autocomplete menu.
+        case "ArrowUp":
+        case "ArrowDown":
+          const firstElement: HTMLButtonElement | null =
+            autocompleteRef.current.querySelector("li > button:not(:disabled)");
+          firstElement && firstElement.focus();
+          event.preventDefault(); // by default, the up and down arrow keys scroll the window
+          break;
+        // the escape key will close the menu and return browser focus to the input.
+        case "Escape":
+          setIsOpen(false);
+          reference.current && reference.current.focus();
+        // the tab, and enter keys will close the menu, and the tab key will move browser
+        // focus forward one element (by default)
+        case "Enter":
+          event.preventDefault();
+          setIsOpen(false);
+          break;
+        case "Tab":
+          setIsOpen(false);
+          break;
+        default:
+          break;
+      }
+      // If the focus is already on the autocomplete menu and the autocomplete menu is open.
+      // hitting tab will close the autocomplete and put browser focus back on the search input.
+    } else if (
+      isOpen &&
+      autocompleteRef.current &&
+      autocompleteRef.current.contains(event.target) &&
+      event.key === "Tab"
+    ) {
+      event.preventDefault();
+      setIsOpen(false);
+      reference.current && reference.current.focus();
+    }
+  };
+
+  const autoCompleteSuggestions = (
+    <Menu ref={autocompleteRef}>
+      <MenuContent>
+        <StyledMenu label="Suggested values" labelHeadingLevel="h3">
+          <MenuList>
+            {autocompleteOptions.map((suggestion) => (
+              <MenuItem
+                aria-label={suggestion}
+                key={suggestion}
+                onClick={(event) =>
+                  handleSuggestionClickInternal(event, suggestion)
+                }
+              >
+                {suggestion}
+              </MenuItem>
+            ))}
+          </MenuList>
+        </StyledMenu>
+      </MenuContent>
+    </Menu>
+  );
+
+  useEffect(() => {
+    const filteredOptions: string[] = suggestions.filter((suggestion) =>
+      suggestion.toLowerCase().includes(filter.toLowerCase()),
+    );
+    setAutocompleteOptions(filteredOptions);
+  }, [filter, suggestions]);
+
+  useEffect(() => {
+    window.addEventListener("keydown", handleMenuKeys);
+    window.addEventListener("click", handleClickOutside);
+    return () => {
+      window.removeEventListener("keydown", handleMenuKeys);
+      window.removeEventListener("click", handleClickOutside);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [isOpen]);
```

Comment:
hmm, is it? The events only need to be added if the component is shown, but I can try without

Reply:
now there is no dependency array, I can't visualize the whole flow, it could be fine, but it could run on every change.

I had empty array in mind. but if that works then it's fine :D

Reply:
true, hmm, it does work though, but I'll add the array with the handlers as dependency.